src/api/context.rs

```diff
@@ -88,6 +88,7 @@ impl<T: Pixel> Context<T> {
   /// let info = FrameParameters {
   ///   frame_type_override: FrameTypeOverride::Key,
   ///   opaque: None,
+  ///   t35_metadata: Box::new([]),
```

Comment:
Better to add default() to FrameParameters.

Reply:
Updated.

I also considered making t35_metadata an option, though it's a bit redundant with an empty list. I think technically the Box::new([]) still needs a malloc, though.

Reply:
Rust designed the containers in its standard library to not allocate memory until data is actually pushed into them. This mean that `Vec::new` and `Box::new([])` don't allocate and thus don't call malloc (or any other functions for that matter, with optimizations turned on).

See this for example: https://godbolt.org/z/Go5GbYvnW

Reply:
Cool. I remember some big debate about that a while back wrt custom allocators but I guess it's not relevant here.